### PR TITLE
Enum plot panel 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by Narcha**
## Introduced Changes

- Multiple synchronized enum plots
- Rounded corners :eyes:

## How to Test

Plot some things

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -90,10 +90,10 @@ control = { path = "crates/control" }
...
 coordinate_systems = { path = "crates/coordinate_systems" }
 ctrlc = { version = "3.2.3", features = ["termination"] }
-eframe = { version = "0.24.1", features = ["persistence"] }
-egui_dock = { version = "0.9.1", features = ["serde"] }
-egui_extras = { version = "0.24.2", features = ["image"] }
-egui_plot = "0.24.1"
+eframe = { git = "https://github.com/Narcha/egui.git", rev="b41419e761b26ac021b538a7a2cd1b79ded90ad9", features = ["persistence"] }
+egui_dock = { git = "https://github.com/Narcha/egui_dock.git", rev="be7758484d937bde68c4130bcebf69c535e7f7f1", features = ["serde"] }
+egui_extras = { git = "https://github.com/Narcha/egui.git", rev="b41419e761b26ac021b538a7a2cd1b79ded90ad9", features = ["image"] }
+egui_plot = { git = "https://github.com/Narcha/egui.git", rev="b41419e761b26ac021b538a7a2cd1b79ded90ad9"}
```
Comment (knoellle) [inline Cargo.toml:96]: Belastend...
Please add a comment here referencing the PR that we're waiting for.

File: tools/twix/src/panels/enum_plot.rs
Comment (knoellle) [inline tools/twix/src/panels/enum_plot.rs:1]: Don't have a particular place to put this, but I have a feature request.
I'd like the panel to automatically switch from Full to Follow mode when zooming and from either of them to Free when panning.
Probably also revert to Full mode when double clicking to reset.

File: Cargo.toml (hunk 1 above)
Comment (Narcha) [reply Cargo.toml:96]: The PR was just merged, maybe we should just wait for the next point release :)

File: tools/twix/src/panels/enum_plot.rs
Comment (Narcha) [reply tools/twix/src/panels/enum_plot.rs:1]: Done :+1:

File: tools/twix/src/panels/enum_plot.rs [hunk 2]
```diff
@@ -74,272 +73,335 @@ impl Segment {
...
+
 pub struct EnumPlotPanel {
     nao: Arc<Nao>,
-    changes: Vec<Change>,
-    messages_count: usize,
-    scroll_position: f64,
-    viewport_width: f64,
-    change_buffer: Option<ChangeBuffer>,
-    output_key: String,
+    segment_datas: Vec<SegmentData>,
```
Comment (knoellle) [inline tools/twix/src/panels/enum_plot.rs:225]: "data" is already plural 😉 
```suggestion
    segment_data: Vec<SegmentData>,
```

**Review by knoellle — CHANGES_REQUESTED** [review]
One minor grammar thing, looks good otherwise.

File: tools/twix/src/panels/enum_plot.rs (hunk 2 above)
Comment (Narcha) [reply tools/twix/src/panels/enum_plot.rs:225]: At least it's consistent with the plot panel. :upside_down_face: 
https://github.com/HULKs/hulk/blob/ca9cb0cedf97f59bc4e2eff577fa367e5061edee/tools/twix/src/panels/plot.rs#L189

Comment (schmidma) [reply tools/twix/src/panels/enum_plot.rs:225]: And that is also incorrect ;)

**Review by knoellle — APPROVED** [review]
GL :tm:

**Pull request merged**